File: src/Moq/Moq.Package/Mock.StaticFactory.cs
namespace Moq
{
    partial class Mock
    {
        static Mock()
        {
            MockFactory.Default = new Sdk.StaticMockFactory();
            OnInitialized();
        }

        /// <summary>
        /// Invoked after the default <see cref="MockFactory.Default"/> 
        /// is initialized.
        /// </summary>
        static partial void OnInitialized();
    }
}

File: src/Moq/Moq.Package/Mock.StaticFactory.vb
Namespace Global.Moq
    Partial Friend Class Mock
        Shared Sub New()
            MockFactory.[Default] = New Sdk.StaticMockFactory
            OnInitialized()
        End Sub

        ''' <summary>
        ''' Invoked after the default <see cref="MockFactory.Default"/> 
        ''' is initialized.
        ''' </summary>
        Partial Private Shared Sub OnInitialized()

        End Sub
    End Class
End Namespace